Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift

```diff
@@ -0,0 +1,200 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import XCTest
+
+@testable import GRPCCodeGen
+
+final class SnippetBasedTranslatorTests: XCTestCase {
+  typealias MethodDescriptor = GRPCCodeGen.CodeGenerationRequest.ServiceDescriptor.MethodDescriptor
+  typealias ServiceDescriptor = GRPCCodeGen.CodeGenerationRequest.ServiceDescriptor
+
+  let echoMethods: [MethodDescriptor] = {
+    var methods: [MethodDescriptor] = []
+    methods.append(
+      MethodDescriptor(
+        documentation: "Immediately returns an echo of a request.",
+        name: "Get",
+        isInputStreaming: false,
+        isOutputStreaming: false,
+        inputType: "Echo_EchoRequest",
+        outputType: "Echo_EchoResponse"
+      )
+    )
+    methods.append(
+      MethodDescriptor(
+        documentation: "Splits a request into words and returns each word in a stream of messages.",
+        name: "Expand",
+        isInputStreaming: false,
+        isOutputStreaming: true,
+        inputType: "Echo_EchoRequest",
+        outputType: "Echo_EchoResponse"
+      )
+    )
+    methods.append(
+      MethodDescriptor(
+        documentation:
+          "Collects a stream of messages and returns them concatenated when the caller closes.",
+        name: "Collect",
+        isInputStreaming: true,
+        isOutputStreaming: false,
+        inputType: "Echo_EchoRequest",
+        outputType: "Echo_EchoResponse"
+      )
+    )
+    methods.append(
+      MethodDescriptor(
+        documentation: "Streams back messages as they are received in an input stream.",
+        name: "Update",
+        isInputStreaming: true,
+        isOutputStreaming: true,
+        inputType: "Echo_EchoRequest",
+        outputType: "Echo_EchoResponse"
+      )
+    )
+    return methods
+  }()
+
+  var echoService: CodeGenerationRequest.ServiceDescriptor {
+    CodeGenerationRequest.ServiceDescriptor(
+      documentation: "An echo service.",
+      name: "Echo",
+      namespace: "echo",
+      methods: echoMethods
+    )
+  }
+
+  var codeGenerationRequest: CodeGenerationRequest {
+    CodeGenerationRequest(
+      fileName: "echo.grpc",
+      leadingTrivia: "Some really exciting license header 2023.",
+      dependencies: [],
+      services: [self.echoService],
+      lookupSerializer: {
+        "ProtobufSerializer<\($0)>()"
+      },
+      lookupDeserializer: {
+        "ProtobufDeserializer<\($0)>()"
+      }
+    )
+  }
+
+  func testTypealiasTranslate() throws {
+    let expectedSwift =
+      """
+      enum echo {
+          enum Echo {
+              enum Collect {
+                  typealias Input = Echo_EchoRequest
+                  typealias Output = Echo_EchoResponse
+                  static let descriptor = MethodDescriptor(
+                      service: echo.Echo,
+                      method: Collect
```

Comment:
This isn't correct, these should be strings:
```suggestion
                      service: "echo.Echo",
                      method: "Collect"
```